deployment/cdk/app.py

```diff
@@ -126,6 +126,8 @@ def __init__(
         self,
         scope: core.Construct,
         stack_id: str,
+        stack_name: str,
```

Comment:
I found the distinction between `stack_id` and `stack_name` less than clear, so there are updates throughout the code for that.

As far as I understand now, `stack_id` is used to name the internal constructs and Cloudformation resources generated by CDK for your deployment, whereas `stack_name` will control the actual Cloudformation stack name of your resources and is generally a more "user friendly" name (e.g. `production-maap-mmt`)